compiler/rustc_mir/src/transform/coverage/spans.rs

```diff
@@ -865,9 +865,9 @@ pub(super) fn filtered_terminator_span(
         // Call `func` operand can have a more specific span when part of a chain of calls
         | TerminatorKind::Call { ref func, .. } => {
             let mut span = terminator.source_info.span;
-            if let mir::Operand::Constant(box constant) = func {
-                if constant.span.lo() > span.lo() {
-                    span = span.with_lo(constant.span.lo());
+            if let mir::Operand::Constant(box (c_span, _)) = func {
```

Comment:
not sure why this special cased to constants. we can do the same thing for other operands by looking up their local's span. But this is essentially what I was talking about for pulling the span even outside the Operand and using (Operand, Span) tuples where necessary

Reply:
I'm not sure I follow your comment, are you referring to something pre-existing? in that case, should we address it in a follow up PR? or are you asking me to change how I'm handling spans further?

Reply:
yea, I'm just musing on the pre-existing thing